### PR TITLE
feat(gridcomponent): developed and used

### PR DESCRIPTION
**Pull request by alebinson**

### CODE REVIEW

File: src/discrete-layer/components/map-container/polygon-selection-ui.spec.tsx [hunk 1]
```diff
@@ -142,17 +138,17 @@ describe('Polygon Selection component', () => {
...
           n.prop('children').props['id'] ===
             'polygon-selection.clear-menu_option.text'
         );
-      })
-      .simulate('click');
+      });
+    item.parent().simulate('click');
 
-    expect(wrapper.find(Menu).prop('open')).toBe(false);
+    // expect(wrapper.find(Menu).prop('open')).toBe(false);
```
Comment (asafMasa) [inline src/discrete-layer/components/map-container/polygon-selection-ui.spec.tsx:151]: ?

File: src/discrete-layer/views/discrete-layer-view.tsx [hunk 2]
```diff
@@ -140,7 +141,10 @@ const DiscreteLayerView: React.FC = observer(() => {
                   <DrawerSubtitle>Subtitle</DrawerSubtitle>
                 </DrawerHeader>
                 <DrawerContent>
-                  <div style={{backgroundColor: 'red', height: '100%', width:'100%'}}></div>
+                  <LayersResultsComponent 
+                    style={{height: '450px',width: '100%'}}
+                  />
+                  {/* <div style={{backgroundColor: 'red', height: '100%', width:'100%'}}></div> */}
```
Comment (asafMasa) [inline src/discrete-layer/views/discrete-layer-view.tsx:147]: why commented?

**Review by asafMasa — CHANGES_REQUESTED** [review]
🥇

File: src/common/components/grid/cell-renderer/details-expander.cell-renderer.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.expanderContainer{
+  margin-top: 4px;
+  margin-left: 8px;
+}
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/common/components/grid/cell-renderer/details-expander.cell-renderer.css:4]: Done

File: src/common/components/grid/grid.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  detailsRowHeight?: number;
+};
+export interface IGridRowDataDetailsExt {
+  rowHeight: number;
+  fullWidth: boolean;
+  isVisible: boolean;
+}
+
+export const GridComponent: React.FC<GridComponentProps> = (props) => {
+  const [gridApi, setGridApi] = useState<GridApi>();
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/common/components/grid/grid.tsx:35]: for feature use, we need it

File: src/common/components/grid/grid.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    columnDefs: [
+      {
+        field: 'isVisible',
+        hide: true,
+      },
+      ...props.gridOptions?.columnDefs as [],
+      props.gridOptions?.detailsRowCellRenderer !== undefined ? 
+        {
+          headerName: '',
+          width: 60,
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/common/components/grid/grid.tsx:55]: Done

File: src/discrete-layer/components/layers-results/cell-renderer/layer-details.cell-renderer.css [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  display: flex;
+  flex-direction: column;
+  row-gap: 4px;
+}
+.detailsProp{
+  padding: 2px 0px;
+  margin: 0px;
+  width: 165px;
+  word-break: break-all;
+}
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/discrete-layer/components/layers-results/cell-renderer/layer-details.cell-renderer.css:24]: Done

File: src/discrete-layer/components/layers-results/layers-results.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { useIntl } from 'react-intl';
+import { GridComponent, GridComponentOptions } from '../../../common/components/grid';
+import { ILayerImage } from '../../models/layerImage';
+import { LayerDetailsRenderer } from './cell-renderer/layer-details.cell-renderer';
+
+const createMockData = (count: number, prefix: string): ILayerImage[] => {
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/discrete-layer/components/layers-results/layers-results.tsx:7]: Done

File: src/discrete-layer/components/layers-results/layers-results.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  ];
+  const gridOptions: GridComponentOptions = {
+    pagination: pagination,
+    paginationPageSize: pageSize,
+    columnDefs: colDef,
+    getRowNodeId: (data: ILayerImage) => {
+      return data.id;
+    },
+    detailsRowCellRenderer: 'detailsRenderer',
+    detailsRowHeight: 150,
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/discrete-layer/components/layers-results/layers-results.tsx:68]: This case no need it's very coupled to column def, from my POV it's Ok

File: src/discrete-layer/components/map-container/polygon-selection-ui.spec.tsx (hunk 1 above)
Comment (alebinson) [reply src/discrete-layer/components/map-container/polygon-selection-ui.spec.tsx:151]: Done

File: src/discrete-layer/views/discrete-layer-view.tsx (hunk 2 above)
Comment (alebinson) [reply src/discrete-layer/views/discrete-layer-view.tsx:147]: It's still no production, helps in dev time, removed anyway

**Review by asafMasa — APPROVED** [review]

**Pull request merged**